lib/std/c/haiku.zig

```diff
@@ -371,6 +372,7 @@ pub const KERN = struct {};
 pub const IOV_MAX = 1024;
 
 pub const PATH_MAX = 1024;
+pub const NAME_MAX = 256;
```

Comment:
NAME_MAX is defined as `length of a file name component, not including the terminating null character`,  so this looks interesting. What is the source?

Reply:
https://xref.landonf.org/source/xref/haiku/headers/posix/limits.h#41

Reply:
Tried to verify this using `mkdir` on a virtual machine and it looks like the real max is 255; it seems they define it as 256 but then use `>=` checks [like this](https://xref.landonf.org/source/xref/haiku/src/system/kernel/fs/vfs.cpp#2048).

![haiku-max](https://user-images.githubusercontent.com/2389051/196058048-a35bf5d1-dd5f-40f2-a023-78424add9194.PNG)

(top is 255 repeated `1`s, bottom is 256)

Reply:
Some things that might change this limit, what API is mkdir using to create the directory (i.e. CreateDirectoryA or CreateDirectoryW)?  Also, what filesystem was this on?  NTFS/FAT/etc?

Reply:
@marler8997 this is HaikuOS in a VM.